src/mine-sweeper.tsx

```diff
@@ -244,6 +244,9 @@ export default class MineSweeper extends React.Component<IMineSweeperProps, IMin
       });
       this.timerId = window.setInterval(this.updateTimer.bind(this), 1000);
     }
+    if (this.isFlagged(x, y)) {
```

Comment:
Great to see that you've added `isFlagged` as helper method. The condition could be part of the early return in line `238` though. I suggest you just add another OR and the isFlagged check as third condition.